### PR TITLE
[Silabs] General updates and fixes for the Silabs Wi-Fi platforms

### PR DESCRIPTION
**Pull request by mkardous-silabs**
#### Description
PR upstreams a few updates and fixes for the different silabs Wi-Fi platforms

- Remove the double retry algorithms in favor on the simili fibanocci algorithm. A follow up PR will be done to clean up the retry logic.
- Rename the ot polling configuration to the transport polling configuration. The goal is for the configuration to be agnostic of the underlying transport
- Update the include call for the spi-multiplex header
- In the previous bit of code, a namespace was not moved with the code that uses the content of the namespace. PR moves the namespace to where it should have been.
- PR fixes the WF200 run time issue where there a silent name conflict between a set of defines and an typedef enum which caused run time failures due to certain files having one value and other files having a second set of values
- Deletes a few lingering todos that are unnecessary and adds a few as placeholders

#### Tests
- CI
- Manual tests

### CODE REVIEW

**Review by lpbeliveau-silabs — APPROVED** [review]

**Review by ksperling-apple — APPROVED** [review]

File: src/platform/silabs/wifi/wf200/platform/wf200_init.c [hunk 1]
```diff
@@ -319,6 +319,7 @@ sl_status_t sl_wfx_host_reset_chip(void)
  *****************************************************************************/
 sl_status_t sl_wfx_host_wait_for_wake_up(void)
 {
+    xSemaphoreTake(wfx_wakeup_sem, pdMS_TO_TICKS(0));
```
Comment (rosahay-silabs) [inline src/platform/silabs/wifi/wf200/platform/wf200_init.c:322]: why was this added ?

**Review by rosahay-silabs — APPROVED** [review]

**Review by senthilku — COMMENTED** [review]
The changes look fine.

File: src/platform/silabs/wifi/wf200/platform/wf200_init.c (hunk 1 above)
Comment (mkardous-silabs) [reply src/platform/silabs/wifi/wf200/platform/wf200_init.c:322]: I had initially removed this because it seemed like an odd thing to do. After double checking the Si SDK file from which this was created from, that is what the platform do so i added the line back to match.
